MobileCenterCrashes/MobileCenterCrashes/MSCrashes.mm

```diff
@@ -789,18 +732,92 @@ - (void)processLogBufferAfterCrash {
+ */
+- (NSArray<MSErrorReport *> *)getUnprocessedCrashReports {
+  dispatch_semaphore_wait(self.delayedProcessingSemaphore, DISPATCH_TIME_FOREVER);
+  dispatch_semaphore_signal(self.delayedProcessingSemaphore);
+  return self.unprocessedReports;
+}
+
+/**
+ * Resumes processing for a list of error reports that is a subset of the unprocessed reports.
+ */
+- (void)sendCrashReportsOrAwaitUserConfirmationForFilteredList:(NSArray<MSErrorReport *> *)filteredList {
+  NSMutableArray *filteredOutLogs = [[NSMutableArray alloc] init];
+  NSMutableArray *filteredOutReports = [[NSMutableArray alloc] init];
+  NSMutableArray *filteredOutFilePaths = [[NSMutableArray alloc] init];
+  for (NSUInteger i = 0; i < [self.unprocessedReports count]; i++) {
+    MSErrorReport *report = self.unprocessedReports[i];
+    MSErrorReport *foundReport = nil;
+    for (MSErrorReport* filteredReport in filteredList) {
+      if ([filteredReport.incidentIdentifier isEqualToString:report.incidentIdentifier]) {
+        foundReport = filteredReport;
+        break;
+      }
+    }
+    
+    // Use the report from the list in case it was modified at all.
+    if (foundReport) {
+      self.unprocessedReports[i] = foundReport;
+    } else {
+      MSAppleErrorLog *log = self.unprocessedLogs[i];
+      NSURL *filePath = self.unprocessedFilePaths[i];
+      [filteredOutReports addObject:report];
+      [filteredOutLogs addObject:log];
+      [filteredOutFilePaths addObject:filePath];
+      
+      // Remove the items from disk.
+      [MSWrapperExceptionManager deleteWrapperExceptionWithUUIDString:report.incidentIdentifier];
+      [self deleteCrashReportWithFileURL:filePath];
+      [self.crashFiles removeObject:filePath];
+    }
+  }
+  
+  // Remove filtered out items from memory.
+  [self.unprocessedLogs removeObjectsInArray:filteredOutLogs];
+  [self.unprocessedFilePaths removeObjectsInArray:filteredOutFilePaths];
+  [self.unprocessedReports removeObjectsInArray:filteredOutReports];
+  
+  // Send or await user confirmation.
+  [self sendCrashReportsOrAwaitUserConfirmation];
+}
+
+/**
+ * Sends error attachments for a particular error report.
+ */
+- (void)sendErrorAttachments:(NSArray<MSErrorAttachmentLog *> *)errorAttachments withIncidentIdentifier:(NSString *)incidentIdentifier {
+  
+  // Send attachements log to log manager.
+  unsigned int totalProcessedAttachments = 0;
+  for (MSErrorAttachmentLog *attachment in errorAttachments) {
+    attachment.errorId = incidentIdentifier;
+    if (![MSCrashes validatePropertiesForAttachment:attachment]) {
+      MSLogError([MSCrashes logTag], @"Not all required fields are present in MSErrorAttachmentLog.");
+      continue;
+    }
+    [self.logManager processLog:attachment forGroupId:self.groupId];
+    ++totalProcessedAttachments;
+  }
+  if (totalProcessedAttachments > kMaxAttachmentsPerCrashReport) {
+    MSLogWarning([MSCrashes logTag], @"A limit of %u attachments per error report might be enforced by server.",
```

Comment:
I suggest logging `totalProcessedAttachments` number as well.

Reply:
I can, but this line existed before this PR, so I don't know if that change belongs in this PR. (I moved the code into a helper method, so it looks like an addition I made.)